lisa/nic.py

```diff
@@ -200,39 +243,36 @@ def get_secondary_nic(self) -> NicInfo:
     def get_nic_by_index(self, index: int = -1) -> NicInfo:
         # get nic by index, default is -1 to give a non-primary nic
         # when there are more than one nic on the system
-        number_of_nics = len(self.get_upper_nics())
+        number_of_nics = len(self.get_nics())
```

Comment:
`get_nics` are used multiple times in this method, it should be saved in a variable.

Reply:
updated